browser/ui/tabs/shared_pinned_tab_service.cc

```diff
@@ -196,7 +198,29 @@ SharedPinnedTabService::SharedPinnedTabService(Profile* profile)
     : profile_(profile) {
   DCHECK(base::FeatureList::IsEnabled(tabs::features::kBraveSharedPinnedTabs));
   profile_observation_.Observe(profile_);
-  browser_list_observation_.Observe(BrowserList::GetInstance());
+
+  shared_pinned_tab_enabled_.Init(
+      brave_tabs::kSharedPinnedTab, profile_->GetPrefs(),
+      base::BindRepeating(
+          [](base::WeakPtr<SharedPinnedTabService> service) {
+            if (UNLIKELY(!service)) {
+              // As BooleanPrefMember is already RAII style subscription,
+              // this is unlikely to happen but theoretically might happen on
```

Comment:
Why do you think it's available in theory?
As `shared_pinned_tab_enabled_` is member var of this service, it'll be destroyed before service is gone.
Also `weak_ptr_factory_` is last declared member, so it should not be null when it's referred by other member vars.

Reply:
Oh wait, shared_pinned_tab_enabled_ can be alive while weak_ptr_factory_ is invalidated, as destruction order is from the last? I think making a method would be clear.

Reply:
Ah, right. it's invalidated at first than all others.